src/OpenSpaceToolkit/Astrodynamics/Trajectory/Orbit.cpp

```diff
@@ -245,11 +247,12 @@ Pass Orbit::getPassWithRevolutionNumber(const Integer& aRevolutionNumber) const
             Integer currentRevolutionNumber = currentPass.isDefined() ? currentPass.getRevolutionNumber()
                                                                       : this->modelPtr_->getRevolutionNumberAtEpoch();
             Instant previousInstant = currentPass.isDefined()
-                                        ? (currentPass.getInterval().accessEnd() + Duration::Microseconds(1.0))
+                                        ? (currentPass.accessInstantAtPassBreak() + Duration::Microseconds(1.0))
```

Comment:
Sanity check:

From our conversation, I thought I remembered that `passBreak` is the instant immediately after the spacecraft crosses the ascending node. If that's true, would this logic make the next pass start one step after the actual start of the pass?

Reply:
`passBreak` is the ascending node (within accuracy). Since `passBreak` is at ascending node, We want to start propagating from a microsecond after so that we don't start before the ascending node.